examples/basic/ipc_shm_ipcapi.sh

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (C) 2024 Intel Corporation
+#
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+
+#!/bin/bash
+
+# port should be a number from the range <1024, 65535>
+PORT=$(( 1024 + ( $$ % ( 65535 - 1024 ))))
+
+# The ipc_shm_ipcapi example requires using pidfd_getfd(2)
+# to obtain a duplicate of another process's file descriptor.
+# Permission to duplicate another process's file descriptor
+# is governed by a ptrace access mode PTRACE_MODE_ATTACH_REALCREDS check (see ptrace(2))
+# that can be changed using the /proc/sys/kernel/yama/ptrace_scope interface.
+PTRACE_SCOPE_FILE="/proc/sys/kernel/yama/ptrace_scope"
+VAL=0
+if [ -f $PTRACE_SCOPE_FILE ]; then
+	PTRACE_SCOPE_VAL=$(cat $PTRACE_SCOPE_FILE)
+	if [ $PTRACE_SCOPE_VAL -ne $VAL ]; then
+		echo "Setting ptrace_scope to 0 (classic ptrace permissions) ..."
+		echo "$ sudo bash -c \"echo $VAL > $PTRACE_SCOPE_FILE\""
+		sudo bash -c "echo $VAL > $PTRACE_SCOPE_FILE"
+	fi
+	PTRACE_SCOPE_VAL=$(cat $PTRACE_SCOPE_FILE)
+	if [ $PTRACE_SCOPE_VAL -ne $VAL ]; then
+		echo "SKIP: setting ptrace_scope to 0 (classic ptrace permissions) FAILED - skipping the test"
+		exit 0
+	fi
+fi
+
+echo "Starting ipc_shm_ipcapi CONSUMER on port $PORT ..."
+./umf_example_ipc_shm_ipcapi_consumer $PORT &
```

Comment:
Instead of using a script to run the example can one process (e.g. consumer) just start the producer process?

Reply:
Such processes are not independent - the child process inherits some stuff from the parent process (AFAIR: environment, open file descriptors, ...). More over we have to set `ptrace_scope` to 0 - it is also done in this script.

Reply:
I think that this way of running processes is closer to the requirements for UMF IPC functionality